packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -106,6 +150,43 @@ export async function assertUserPoolResource(projRoot: string, gen1UserPoolId: s
   expect(gen2Resource).toEqual(gen1Resource);
 }
 
+export async function assertStorageResource(projRoot: string, gen1BucketName: string, gen1Region: string) {
+  const gen1Resource = await getResourceDetails('AWS::S3::Bucket', gen1BucketName, gen1Region);
+  removeProperties(gen1Resource, ['DualStackDomainName', 'DomainName', 'BucketName', 'Arn', 'RegionalDomainName', 'Tags', 'WebsiteURL']);
+  // TODO: remove below line after CorsConfiguration.CorsRules[0].Id inconsistency is fixed
+  removeProperties(gen1Resource, ['CorsConfiguration.CorsRules[0].Id']);
+
+  const gen2Meta = getProjectOutputs(projRoot);
+  const gen2BucketName = gen2Meta.storage.bucket_name;
+  const gen2Region = gen2Meta.storage.aws_region;
+  const gen2Resource = await getResourceDetails('AWS::S3::Bucket', gen2BucketName, gen2Region);
+  removeProperties(gen2Resource, ['DualStackDomainName', 'DomainName', 'BucketName', 'Arn', 'RegionalDomainName', 'Tags', 'WebsiteURL']);
+
+  expect(gen2Resource).toEqual(gen1Resource);
+}
+
+export async function assertFunctionResource(projRoot: string, gen1FunctionName: string, gen1Region: string) {
+  const gen1Resource = await getResourceDetails('AWS::Lambda::Function', gen1FunctionName, gen1Region);
+  removeProperties(gen1Resource, ['Arn', 'FunctionName', 'LoggingConfig.LogGroup', 'Role']);
+  // TODO: remove below line after RecursiveLoop, RuntimeManagementConfig, Tags is fixed
+  removeProperties(gen1Resource, ['RecursiveLoop', 'RuntimeManagementConfig', 'Tags']);
```

Comment:
I assume we have a task to follow up created?

Reply:
Yes, we have it created under QA issues